### PR TITLE
Add timelimit for extention execution

### PR DESCRIPTION
**Pull request by nati**
We need to stop extention VM if it runs too long.
(eg infinite loop bug)

Closes issue: #54

### CODE REVIEW

File: extension/framework/runner/environment.go [hunk 1]
```diff
@@ -73,7 +74,7 @@ func (env *Environment) InitializeEnvironment() error {
 	if err != nil {
 		return fmt.Errorf("Failed to connect to database: %s", err.Error())
 	}
-	env.Environment = gohan_otto.NewEnvironment(env.dbConnection, &middleware.FakeIdentity{})
+	env.Environment = gohan_otto.NewEnvironment(env.dbConnection, &middleware.FakeIdentity{}, time.Duration(30)*time.Second)
```
Comment (kwapik) [inline extension/framework/runner/environment.go]: AFAIK you don't need to wrap `30` in `time.Duration()`, simple `30 * time.Second` will do the cast.

File: extension/otto/gohan_db_test.go [hunk 2]
```diff
@@ -29,6 +30,13 @@ import (
 )
 
 var _ = Describe("GohanDb", func() {
+	var (
+		timelimit time.Duration
+	)
+
+	BeforeEach(func() {
+		timelimit = time.Duration(1) * time.Second
```
Comment (kwapik) [inline extension/otto/gohan_db_test.go]: ditto

File: extension/otto/otto_test.go [hunk 3]
```diff
@@ -38,11 +38,13 @@ var _ = Describe("Otto extension manager", func() {
 	var (
 		manager            *schema.Manager
 		environmentManager *extension.Manager
+		timelimit          time.Duration
 	)
 
 	BeforeEach(func() {
 		manager = schema.GetManager()
 		environmentManager = extension.GetManager()
+		timelimit = time.Duration(1) * time.Second
```
Comment (kwapik) [inline extension/otto/otto_test.go]: ditto

File: server/extension.go [hunk 4]
```diff
@@ -36,8 +36,9 @@ import (
 	"github.com/cloudwan/gohan/extension"
 	"github.com/cloudwan/gohan/extension/otto"
 	"github.com/cloudwan/gohan/server/middleware"
+	"time"
 )
 
-func newEnvironment(dataStore db.DB, identity middleware.IdentityService) extension.Environment {
-	return otto.NewEnvironment(dataStore, identity)
+func newEnvironment(dataStore db.DB, identity middleware.IdentityService, timelimit int) extension.Environment {
+	return otto.NewEnvironment(dataStore, identity, time.Duration(timelimit)*time.Second)
```
Comment (kwapik) [inline server/extension.go]: ditto

File: util/config.go [hunk 5]
```diff
@@ -95,6 +95,16 @@ func (config *Config) GetString(key, defaultValue string) string {
 	return dataString
 }
 
+//GetInt returns int parameter from config
+func (config *Config) GetInt(key string, defaultValue int) int {
+	data := config.GetParam(key, defaultValue)
+	dataInt, ok := data.(int)
+	if ok == false {
```
Comment (kwapik) [inline util/config.go]: Why not `if !ok`?

File: extension/framework/runner/environment.go (hunk 1 above)
Comment (nati) [inline extension/framework/runner/environment.go]: done

File: extension/otto/gohan_db_test.go (hunk 2 above)
Comment (nati) [inline extension/otto/gohan_db_test.go]: done

File: extension/otto/otto_test.go (hunk 3 above)
Comment (nati) [inline extension/otto/otto_test.go]: done

File: server/extension.go (hunk 4 above)
Comment (nati) [inline server/extension.go]: No this cast won't work

File: util/config.go (hunk 5 above)
Comment (nati) [inline util/config.go]: done

**Pull request merged**